cmd/init.go

```diff
@@ -80,3 +111,29 @@ Next steps:
 	logger.Infof(successMsg, configPath, configPath)
 	return nil
 }
+
+func createClusterConfigFromTemplate(outputFilePath string, templateOpts interface{}, fileTemplate string) error {
```

Comment:
My intention to keep this within the `filegen` package was to make some momentum to write unit tests for it. We don't usually write tests for anything under the `cmd` package in the golang world. Am I right?

Reply:
I would disagree with this approach for 2 reasons.

One would be for scope. Being that this method is only used in a single place AND it's `cluster.yaml` specific, we can and should restrict its scope as much as possible.

The second reason is that there's nothing stopping us writing tests for the `cmd` package. We shouldn't be testing the commands themselves as that could result in us testing the library we use for commands, in our case Cobra. But nothing stops us from having a test file that tests the methods that make up the command if it's declared in the `cmd` package.

I guess the compromise would be to keep the `filegen` package but make the method not `cluster.yaml` specific and then call the method from the init command with `cluster.yaml` as a parameter.

LMK what you think @mumoshu

Reply:
> 
this method is only used in a single place AND it's `cluster.yaml` specific, we can and should restrict its scope as much as possible.

That's a good point!

> 
But nothing stops us from having a test file that tests the methods that make up the command if it's declared in the `cmd` package.

True.

Also - I just noticed I had hard-coded `"cluster.yaml"` in the original `filegen/filegen.go`. What a non-sense!

> 
I guess the compromise would be to keep the `filegen` package but make the method not `cluster.yaml` specific and then call the method from the init command with `cluster.yaml` as a parameter

But it is still used only from `init.go`, right? Then I'm fine with your current work.

Reply:
@mumoshu Let me know what you think of my latest change. I've made the templating more generic and seperating the concerns of creating a file and rendering the template into different layers